### PR TITLE
DMP-2205 - add pending retention to auto closed cases.

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
https://tools.hmcts.net/jira/browse/DMP-2205

### CODE REVIEW

**Review by mestebanez — COMMENTED** [review]
did you mean to checkin .idea/codeStyles/Project.xml

File: src/main/java/uk/gov/hmcts/darts/retention/service/impl/RetentionPostServiceImpl.java [hunk 1]
```diff
@@ -165,23 +165,30 @@ private CaseRetentionEntity getLatestCompletedCaseRetention(CourtCaseEntity cour
 
     private CaseRetentionEntity createNewCaseRetention(PostRetentionRequest postRetentionRequest, CourtCaseEntity courtCase,
```
Comment (mestebanez) [inline src/main/java/uk/gov/hmcts/darts/retention/service/impl/RetentionPostServiceImpl.java:166]: Add private below public

File: src/main/java/uk/gov/hmcts/darts/cases/service/impl/CloseOldCasesProcessorImpl.java [hunk 2]
```diff
@@ -73,24 +84,34 @@ private void closeCase(CourtCaseEntity courtCase) {
...
-    private void closeCaseInDb(CourtCaseEntity courtCase, OffsetDateTime caseClosedDate) {
+    private void closeCaseInDbAndAddRetention(CourtCaseEntity courtCase, OffsetDateTime caseClosedDate) {
         courtCase.setClosed(TRUE);
         courtCase.setCaseClosedTimestamp(caseClosedDate);
         caseRepository.save(courtCase);
+
+        LocalDate retentionDate = retentionDateHelper.getRetentionDateForPolicy(courtCase, RetentionPolicyEnum.DEFAULT);
+
+        PostRetentionRequest postRetentionRequest = new PostRetentionRequest();
+        postRetentionRequest.setComments("CloseOldCases Automated job setting retention period to Default");
```
Comment (mestebanez) [inline src/main/java/uk/gov/hmcts/darts/cases/service/impl/CloseOldCasesProcessorImpl.java:110]: Maybe add the string as a class constant

**Comment by cmagowan-hmcts** [comment]
> did you mean to checkin .idea/codeStyles/Project.xml

yes, the default intellij imports sorting is sorting them in a different order from what checkstyles is expecting and raising issues

File: src/main/java/uk/gov/hmcts/darts/retention/service/impl/RetentionPostServiceImpl.java (hunk 1 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/retention/service/impl/RetentionPostServiceImpl.java]: done

File: src/main/java/uk/gov/hmcts/darts/cases/service/impl/CloseOldCasesProcessorImpl.java (hunk 2 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/cases/service/impl/CloseOldCasesProcessorImpl.java]: done

**Review by mestebanez — APPROVED** [review]

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**